### ISSUE TITLE
Temporary performance drop on `1.4.0-alpha1` benchmark

### ISSUE THREAD

**Issue by oleschoenburg** [camunda/zeebe#8839]
**Description**

The long-running benchmark for `1.4.0-alpha1` had a sudden, quite severe performance drop.
There appers to be no leader changes, and no pod restarts during that time period.

Logs here: https://drive.google.com/file/d/1-JyI21z0qb_Y2KOVjDhHBVtPY_aMy8po/view?usp=sharing

![Screenshot 2022-02-25 at 10-22-58 Zeebe - Grafana](https://user-images.githubusercontent.com/1379753/155702248-7d30f3c6-e21d-4251-99c7-a956c63cc071.png)
![Screenshot 2022-02-25 at 10-24-02 Zeebe - Grafana](https://user-images.githubusercontent.com/1379753/155702259-b6147d6b-a182-44e2-be7a-176a23bc02e2.png)
![Screenshot 2022-02-25 at 10-24-20 Zeebe - Grafana](https://user-images.githubusercontent.com/1379753/155702262-2ff00d85-821b-4587-8d2a-2ebe24854a15.png)

**Comment by menski** [comment]
@deepthidevaki as you are medic this week, it would be a possible topic to investigate the situation

**Comment by deepthidevaki** [comment]
**TLDR:** Performance drop is due to high disk queue length which significantly adds delay to disk operations.


Zeebe-0 is the leader during the performance drop. 

* We can see that the Segment flush duration is higher than normal during this time. 
![image](https://user-images.githubusercontent.com/1997478/156146567-adedec90-a99c-44c6-adf8-7313b5620878.png)
Avg calculated using the following query: 
`sum(rate(atomix_segment_flush_time_sum{namespace=~"$namespace",pod=~"$pod", partition=~"$partition"}[1m])) by (pod) / sum(rate(atomix_segment_flush_time_count{namespace=~"$namespace",pod=~"$pod", partition=~"$partition"}[1m])) by (pod)`

* Cpu usage of zeebe-0 looks normal:
![image](https://user-images.githubusercontent.com/1997478/156146715-26238e1c-c782-4e43-9016-ab73cc4325de.png)


* So I looked at google cloud monitoring to see if the disk operations are throttled, but observed that throttling rate is lower during this period. But found that `queue_length` is unusually high during this period. (Following is the metric from the node where zeebe-0 was running at that time). 
![image](https://user-images.githubusercontent.com/1997478/156147397-fc14a7f3-20a9-4f18-b8ca-8becb31be41d.png)
"The queue length on the disk averaged over the last 60 seconds. For Container-Optimized OS, or Ubuntu running GKE."

* This matches with the performance drop: 
Number of disk operations have reduced
![image](https://user-images.githubusercontent.com/1997478/156147467-8538b4fa-c379-498d-9cc1-ee804ec350a5.png)

and the throughput drops: 
![image](https://user-images.githubusercontent.com/1997478/156147590-225f73ed-78cd-41c1-8bf1-47efeeb81824.png)

It is not clear why disk queue length is suddenly high. I don't know yet if it is something influenced by Zeebe or the infrastructure.

**Comment by deepthidevaki** [comment]
We can see a similar pattern at other times:

Performance drop between 15:30-15:55 (zeebe-1 is the leader)
![image](https://user-images.githubusercontent.com/1997478/156150053-da9019b3-5660-4007-beb5-9fb17a4a709f.png)
![image](https://user-images.githubusercontent.com/1997478/156150185-a45a1604-0efa-4286-8999-75b91d91174e.png)
Queue length
![image](https://user-images.githubusercontent.com/1997478/156150238-b833e662-7f5b-4e21-8744-beff5eddaa9c.png)

**Comment by npepinpe** [comment]
I will close this for now - we should have an actual way of monitoring performance by the end of the quarter, and we'll focus on that instead.
